### ISSUE TITLE
Use compliment for Clojure REPL completions

### ISSUE THREAD

**Issue by arichiardi** [clojure-emacs/inf-clojure#111]
It looks like what `inf-clojure` currently uses for completions is broken.

Cider uses the wonderful https://github.com/alexander-yakushev/compliment and we probably should as well. It gives you more completion power as well because it accepts a context in input, which it then uses to complete locals, for instance.

It also returns more metadata together with the actual completion and we could feed that to `company-mode`.

**Comment by bbatsov** [comment]
That's because REPLy uses the old completion library internally and it's always available to users of boot and leiningen (which use REPLy). The only real fix for this is to finish somehow https://github.com/trptcolin/reply/pull/153

**Comment by arichiardi** [comment]
I see, I opened this because I had some exception and I basically realized that according to your message I don't have any completion in `boot` when using a socket repl. That's where the exception comes from.

**Comment by bbatsov** [comment]
When you can easily add a check for the presence of compliment and use different completion commands for it. That'd be simple approach to solve this problem.

**Comment by arichiardi** [comment]
Compliment sends back edn. So I am thinking we could have a var containing a function which is called with whatever the completion command returns. This function would be user provided (with a default), to that the passing of the compliment's edn can be done (potentially using `parseclj`.

**Comment by arichiardi** [comment]
Or a different module `inf-clojure-company` can provide this function for `company-mode` for instance.

**Comment by bbatsov** [comment]
> Compliment sends back edn. So I am thinking we could have a var containing a function which is called with whatever the completion command returns. This function would be user provided (with a default), to that the passing of the compliment's edn can be done (potentially using parseclj.

Yeah, that sounds like a good approach to me. edn.el (https://github.com/expez/edn.el) can also be used (at least temporary, until `parseclj` gets a stable release).

**Comment by arichiardi** [comment]
I am working on a solution for this, it will require `compliment` on the classpath.

**Comment by MalloZup** [comment]
autogenerated with https://github.com/MalloZup/doghub:  issue inactive since 450 days. Please update the issue or close it
